src/components/Message/MessageLivestream.js

```diff
@@ -1,260 +1,366 @@
-    isMyMessage: PropTypes.func,
-    /**
-     * Returns all allowed actions on message by current user e.g., [edit, delete, flag, mute]
-     * Please check [Message](https://github.com/GetStream/stream-chat-react/blob/master/src/components/Message.js) component for default implementation.
-     * */
-    getMessageActions: PropTypes.func,
-    /**
-     * Function to publish updates on message to channel
-     *
-     * @param message Updated [message object](https://getstream.io/chat/docs/#message_format)
-     * */
-    updateMessage: PropTypes.func,
-    /**
-     * Reattempt sending a message
-     * @param message A [message object](https://getstream.io/chat/docs/#message_format) to resent.
-     */
-    handleRetry: PropTypes.func,
-    /**
-     * Add or remove reaction on message
-     *
-     * @param type Type of reaction - 'like' | 'love' | 'haha' | 'wow' | 'sad' | 'angry'
-     * @param event Dom event which triggered this function
-     */
-    handleReaction: PropTypes.func,
-    /** If actions such as edit, delete, flag, mute are enabled on message */
-    actionsEnabled: PropTypes.bool,
-    /** DOMRect object for parent MessageList component */
-    messageListRect: PropTypes.object,
-    /**
-     * Handler for actions. Actions in combination with attachments can be used to build [commands](https://getstream.io/chat/docs/#channel_commands).
-     *
-     * @param name {string} Name of action
-     * @param value {string} Value of action
-     * @param event Dom event that triggered this handler
-     */
-    handleAction: PropTypes.func,
-    /**
-     * The handler for hover event on @mention in message
-     *
-     * @param event Dom hover event which triggered handler.
-     * @param user Target user object
-     */
-    onMentionsHoverMessage: PropTypes.func,
-    /**
-     * The handler for click event on @mention in message
-     *
-     * @param event Dom click event which triggered handler.
-     * @param user Target user object
-     */
-    onMentionsClickMessage: PropTypes.func,
-    /**
-     * The handler for click event on the user that posted the message
-     *
-     * @param event Dom click event which triggered handler.
-     */
-    onUserClick: PropTypes.func,
-    /**
-     * The handler for mouseOver event on the user that posted the message
-     *
-     * @param event Dom mouseOver event which triggered handler.
-     */
-    onUserHover: PropTypes.func,
-    /**
-     * The component that will be rendered if the message has been deleted.
-     * All of Message's props are passed into this component.
-     */
-    MessageDeleted: PropTypes.elementType,
-  };
-
-  static defaultProps = {
-    Attachment: DefaultAttachment,
-  };
-
-  state = {
-    actionsBoxOpen: false,
-    reactionSelectorOpen: false,
-  };
-
-  reactionSelectorRef = React.createRef();
-
-  editMessageFormRef = React.createRef();
+// eslint-disable-next-line sonarjs/cognitive-complexity
```

Comment:
@jaapbakker88 @vishalnarkhede I'd suggest disabling this rule globally since it's too good to be respected